shared/utils/domains.test.ts

```diff
@@ -1,149 +1,171 @@
-import { stripSubdomain, parseDomain, isCustomSubdomain } from "./domains";
+import { stripSubdomain, parseDomain, isHostedSubdomain } from "./domains";
 // test suite is based on subset of parse-domain module we want to support
 // https://github.com/peerigon/parse-domain/blob/master/test/parseDomain.test.js
 describe("#parseDomain", () => {
+  beforeEach(() => {
+    process.env.URL = "https://example.com";
+  });
+
   it("should remove the protocol", () => {
     expect(parseDomain("http://example.com")).toMatchObject({
       subdomain: "",
-      domain: "example",
-      tld: "com",
+      domain: "example.com",
     });
     expect(parseDomain("//example.com")).toMatchObject({
       subdomain: "",
-      domain: "example",
-      tld: "com",
+      domain: "example.com",
     });
     expect(parseDomain("https://example.com")).toMatchObject({
       subdomain: "",
-      domain: "example",
-      tld: "com",
+      domain: "example.com",
     });
   });
 
   it("should remove sub-domains", () => {
     expect(parseDomain("www.example.com")).toMatchObject({
       subdomain: "www",
-      domain: "example",
-      tld: "com",
+      domain: "example.com",
     });
   });
 
   it("should remove the path", () => {
     expect(parseDomain("example.com/some/path?and&query")).toMatchObject({
       subdomain: "",
-      domain: "example",
-      tld: "com",
+      domain: "example.com",
     });
     expect(parseDomain("example.com/")).toMatchObject({
       subdomain: "",
-      domain: "example",
-      tld: "com",
+      domain: "example.com",
     });
   });
 
   it("should remove the query string", () => {
     expect(parseDomain("example.com?and&query")).toMatchObject({
       subdomain: "",
-      domain: "example",
-      tld: "com",
+      domain: "example.com",
     });
   });
 
   it("should remove special characters", () => {
     expect(parseDomain("http://m.example.com\r")).toMatchObject({
       subdomain: "m",
-      domain: "example",
-      tld: "com",
+      domain: "example.com",
     });
   });
 
   it("should remove the port", () => {
     expect(parseDomain("example.com:8080")).toMatchObject({
       subdomain: "",
-      domain: "example",
-      tld: "com",
+      domain: "example.com",
     });
   });
 
   it("should allow @ characters in the path", () => {
     expect(parseDomain("https://medium.com/@username/")).toMatchObject({
       subdomain: "",
-      domain: "medium",
-      tld: "com",
-    });
-  });
-
-  it("should also work with three-level domains like .co.uk", () => {
-    expect(parseDomain("www.example.co.uk")).toMatchObject({
-      subdomain: "www",
-      domain: "example",
-      tld: "co.uk",
+      domain: "medium.com",
     });
   });
 
   it("should not include private domains like blogspot.com by default", () => {
     expect(parseDomain("foo.blogspot.com")).toMatchObject({
-      subdomain: "foo",
-      domain: "blogspot",
-      tld: "com",
+      subdomain: "",
```

Comment:
This behavior change is pretty worrying, maybe these tests just don't make sense any more?

Reply:
we should talk this one through live, I'll ping once I've cleaned the PR up and we can review some of the stranger corners